src/aerospike_clj/utils.clj

```diff
@@ -11,3 +12,45 @@
   (-> (hosts->cluster hosts)
       (s/split  #"\.")
       first))
+
+(def ^:private boolean-replacements
+  "For bins, Aerospike converts `true` to `1`, `false` to `0` and `nil` values are
+  not accepted. The conversion happens only at the top-level of the bin and does not
+  seem to affect nested structures. To store these values in the database, they are
+  converted to keywords."
+  {true  :true
+   false :false
+   nil   :nil})
+
+;; transducers
+(def ^:private x-sanitize
+  (replace boolean-replacements))
+
+(def ^:private x-desanitize
+  (replace (set/map-invert boolean-replacements)))
+
+;; predicates
+(defn single-bin?
+  "Predicate function to determine whether data will be stored as a single bin or
+  multiple bin record."
+  [bin-names]
+  (= bin-names [""]))
+
+(defn string-keys?
+  "Predicate function to determine whether all keys provided for bins are strings."
+  [bin-names]
+  (every? string? bin-names))
+
+(defn sanitize-bin-value
+  "Values in nested structures are unaffected and do not need to be sanitized. If,
+  however, `true`, `false` or `nil` exist as the only value in a bin, they need to
+  be sanitized."
+  [bin-value]
```

Comment:
If `bin-value` is a single (not a sequence) always, why do we need a transducer, to begin with? I think this is my fault for not understanding this earlier. If I understand correctly you use `replace` here  (via transducing) but you never use it on a sequence of bins. So, for a single bin should we could just replace a key with a value from `boolean-replacements` (or the invert of it)?

So, I think `x-bin-convert` is good but `x-sanitize` and `x-desanitize` should not be transducers because they are always 'iterating' over a single bin. WDYT? It might be worth it to [micro benchmark](https://github.com/hugoduncan/criterium) this.

Reply:
After benchmarks, here are some quick stats:

> sanitize-bin-value **with transducer**
`Execution time mean : 266.958899 ns`

> desanitize-bin-value **with transducer**
`Execution time mean : 218.256747 ns`

> sanitize-bin-value
`Execution time mean : 358.403094 ns`

> desanitize-bin-value
`Execution time mean : 403.246815 ns`

Seems like transducers do help very slightly _(100-185 ns faster)_. It's negligible but I will leave transducers in for now unless you feel they are overkill and want to remove them.